scripts/sst_prep_obs.sh

```diff
@@ -1,95 +1,132 @@
-      "sst.avhrr19_l3u.nesdis")
-         subsample=true
-         skip=250
-         ;;
-      "sst.windsat_l3u.ghrsst")
-         subsample=true
-         skip=5
-         ;;
-   esac
+   echo
+   echo Applying THINNING
    echo subsample=$subsample
    echo skip=$skip
-
+   echo
    if [ $subsample ]; then
-      echo "Subsampling $sst_source SST"
+      echo "Subsampling $SSTsource SST"
       # TODO: Subsample elsewhere
-      mv ${PROCobs} ${ObsRunDir}/ioda.sst.${sst_source}_LARGE.nc
+      mv ${PROCobs} ${ObsRunDir}/ioda.sst.${SSTsource}_LARGE.nc
 
       # Create record dim
-      ncks --mk_rec_dmn nlocs ${ObsRunDir}/ioda.sst.${sst_source}_LARGE.nc ${ObsRunDir}/sst-tmp.nc
+      ncks --mk_rec_dmn nlocs ${ObsRunDir}/ioda.sst.${SSTsource}_LARGE.nc ${ObsRunDir}/sst-tmp.nc
       # Subsample
       ncks -F -d nlocs,1,,$skip ${ObsRunDir}/sst-tmp.nc ${PROCobs}
       # Cleanup
       rm ${ObsRunDir}/sst-tmp.nc
-      rm ${ObsRunDir}/ioda.sst.${sst_source}_LARGE.nc   
+      rm ${ObsRunDir}/ioda.sst.${SSTsource}_LARGE.nc   
    fi
 
-   exit
+   return
+fi
+# Check if the raw observations exist and process.
+if [ "$SSTsource" == "sst.avhrr19_l3u.nesdis" ] || \
+   [ "$SSTsource" == "sst.avhrrmta_l3u.nesdis" ]; then
+   SSTsource="sst.avhrr_l3u.nesdis"
+   OBSDCOM=$DCOM_ROOT/${SSTsource}/$PDY              #FullPath of raw obs
+else
+   OBSDCOM=$DCOM_ROOT/${SSTsource}/$PDY              #FullPath of raw obs
 fi
-
-OBSDCOM=$DCOM_ROOT/${sst_source}/$PDY
 if [ -d "$OBSDCOM" ]; then
    
-   OUTDIR=${ROTDIR}/${CDATE}
-   mkdir -p ${OUTDIR}
-   
    cd $OBSDCOM
-   echo SST Observations from ${sst_source} for $PDY exist at `pwd`
-   
+   echo SST Observations from ${SSTsource}-${sat} for $PDY exist and will be processed, obs directory: `pwd` 
+
    s="${IODA_EXEC}/gds2_sst2ioda.py -i "
-   for files in `ls ${filename}*.nc`; do
+   for files in `ls *${sat}*.nc`; do
       s+=" $OBSDCOM/${files} "
    done
    
-   s+=" -o ${OUTDIR}/ioda.${sst_source}.${DA_SLOT_LEN}h.nc -d ${CDATE}"
-
+   s+=" -o ${PREPROCobs} -d ${CDATE}"
    eval ${s}
 
+   echo PreProcessed Observations for ${SSTsource}-${sat} are copied from "${PREPROCobs}" \
+        to ${PROCobs}
+
+   cp -rp ${PREPROCobs} ${PROCobs}
+   echo
+   echo Applying THINNING
+   echo subsample=$subsample
+   echo skip=$skip
+   echo
+   if [ $subsample ]; then
+      echo "Subsampling $SSTsource ${sat} SST"
+      # TODO: Subsample elsewhere
+      mv ${PROCobs} ${ObsRunDir}/ioda.sst.${SSTsource}_LARGE.nc
+
+      # Create record dim
+      ncks --mk_rec_dmn nlocs ${ObsRunDir}/ioda.sst.${SSTsource}_LARGE.nc ${ObsRunDir}/sst-tmp.nc
+      # Subsample
+      ncks -F -d nlocs,1,,$skip ${ObsRunDir}/sst-tmp.nc ${PROCobs}
+      # Cleanup
+      rm ${ObsRunDir}/sst-tmp.nc
+      rm ${ObsRunDir}/ioda.sst.${SSTsource}_LARGE.nc   
+   fi 
 else
    
    set -x
-   echo There are no SST observations from ${sst_source} for ${CDATE}  
+   echo There are no SST observations from ${SSTsource}-${sat} for ${PDY}  
```

Comment:
That's a dash (-) and not an underscore (_), let's keep the one of the two (_).

Reply:
The dash {-} is to separate the sat_l3u.nesdisellite name {AVHRR19, AVHRRMTA, etc) from the SSTsource, which is sst.avhrr_l3u.nesdis